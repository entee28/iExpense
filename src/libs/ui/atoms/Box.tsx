import React from 'react'
import { View, ViewProps, ViewStyle } from 'react-native'

type Props = {
  children?: React.ReactNode
  viewProps?: ViewProps
  testID?: string
} & ViewStyle

export const Box = ({ children, viewProps, testID, ...styles }: Props) => {
  return (
    <View testID={testID} {...viewProps} style={styles}>
      {children}
    </View>
  )
}
