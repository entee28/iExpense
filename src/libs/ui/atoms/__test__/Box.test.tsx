import { render } from '@testing-library/react-native'
import React from 'react'
import { Text } from 'react-native'
import { Box } from '../Box'

it('renders children components correctly', () => {
  const box = render(
    <Box>
      <Text testID="text_children">Hello World</Text>
    </Box>
  )

  const text = box.getByTestId('text_children')
  expect(text.children).toContain('Hello World')
})

it('renders styles correctly', () => {
  const box = render(
    <Box width={300} height={300} backgroundColor="red" testID="red_box" />
  )

  expect(box.getByTestId('red_box')).toHaveStyle({
    width: 300,
    height: 300,
    backgroundColor: 'red'
  })
})

it('renders correctly', () => {
  const snapshot = render(
    <Box
      flexDirection="row"
      alignItems="center"
      justifyContent="center"
      backgroundColor="red">
      <Text>Hello World</Text>
    </Box>
  ).toJSON()

  expect(snapshot).toMatchSnapshot()
})
