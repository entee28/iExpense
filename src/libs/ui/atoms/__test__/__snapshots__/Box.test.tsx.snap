// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello World
  </Text>
</View>
`;
